sdk/identity/azure-identity/azure/identity/_internal/proactive_refresh.py

```diff
@@ -0,0 +1,73 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import abc
+import logging
+import time
+from typing import TYPE_CHECKING
+
+from .._constants import DEFAULT_REFRESH_OFFSET, DEFAULT_TOKEN_REFRESH_RETRY_DELAY
+
+try:
+    ABC = abc.ABC
+except AttributeError:  # Python 2.7, abc exists, but not ABC
+    ABC = abc.ABCMeta("ABC", (object,), {"__slots__": ()})  # type: ignore
+
+if TYPE_CHECKING:
+    # pylint:disable=ungrouped-imports,unused-import
+    from typing import Any, Optional
+    from azure.core.credentials import AccessToken
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ProactiveRefresh(ABC):
```

Comment:
nit: This classname seems a bit misleading since it will not proactively refresh anything unless a request is made. Proactive makes it seem like it will refresh regardless of whether a token was requested.

Reply:
Suggestions welcome 😉